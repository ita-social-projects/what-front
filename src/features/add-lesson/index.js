export { AddLesson } from './add-lesson';
