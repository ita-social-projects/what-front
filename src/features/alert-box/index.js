export { AlertBox } from './alert-box.js';
export * from './redux';
