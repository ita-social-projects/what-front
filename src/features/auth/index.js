export { Auth } from './auth.js';