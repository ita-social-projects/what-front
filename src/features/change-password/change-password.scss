@import '@/styles/main';

.wrapper {
  max-width: 750px;
  margin: 0 auto;
  @include franklin-regular(); 
}