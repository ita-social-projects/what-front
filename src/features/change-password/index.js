export { ChangePassword } from './change-password.js';