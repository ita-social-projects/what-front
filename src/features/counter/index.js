export { Counter } from './counter';
export * from './redux';