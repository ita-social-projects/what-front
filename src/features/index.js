export * from './counter';
export * from './support';
export * from './schedule-student';
export * from './auth';
export * from './header';
export * from './not-found';
export * from './modal-window';
export * from './unassigned-list';
export * from './mentors';
export * from './courses';
export * from './groups';
export * from './lessons';
export * from './secretaries';
export * from './students';
export * from './registration';