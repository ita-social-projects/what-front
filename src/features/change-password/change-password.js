import React, { useEffect, useState } from 'react';
import { shallowEqual, useSelector } from 'react-redux';
import { useHistory } from 'react-router-dom';
import { paths, useActions } from '@/shared';
import { changePasswordSelector, currentUserSelector, newPassword } from '@/models';

import { Button, WithLoading } from '@/components';
import { addAlert, ModalWindow } from '@/features';

import { Formik, Form, Field } from 'formik';
import { changePasswordValidation } from '@features/validation/validation-helpers.js';

import className from 'classnames';
import styles from './change-password.scss';

export const ChangePassword = () => {
  const { currentUser } = useSelector(currentUserSelector, shallowEqual);
  const [password, setPassword] = useState({});

  const {
    isLoading: isChangePasswordLoading,
    isLoaded: isChangePasswordLoaded,
    error: changePasswordError,
  } = useSelector(changePasswordSelector, shallowEqual);

  const [setNewPassword, dispatchAddAlert] = useActions([newPassword, addAlert]);

  const history = useHistory();

  const [toShowModal, setShowModal] = useState(false);

  const handleShowModal = () => setShowModal(true);
  const handleCloseModal = () => setShowModal(false);

  useEffect(() => {
    if (!changePasswordError && isChangePasswordLoaded) {
      history.push(paths.MY_PROFILE);
      dispatchAddAlert('The password has been successfully changed', 'success');
    }
    if (changePasswordError && !isChangePasswordLoaded) {
      dispatchAddAlert(changePasswordError);
    }
  }, [changePasswordError, isChangePasswordLoaded, history, dispatchAddAlert]);

  const onSubmit = (value) => {
    handleShowModal();
    setPassword(value);
  };

  const handleConfirm = () => {
    handleCloseModal();
    setNewPassword(password);
  };

  return (
    <div className={styles.wrapper}>
      <div className="container card shadow col-10">
        <div className="container pb-2">
          <h3 className="pt-3">Change Password</h3>
          <hr />
          <WithLoading isLoading={isChangePasswordLoading && !isChangePasswordLoaded} className="d-block mx-auto">
            <Formik
              initialValues={{
                email: currentUser?.email,
                currentPassword: '',
                newPassword: '',
                confirmNewPassword: '',
              }}
              validationSchema={changePasswordValidation}
              onSubmit={onSubmit}
            >
              {({ values, errors, touched, isValid, dirty }) => (
                <Form>
                  <div className="container px-0">
                    <div className="row m-0">
                      <div className="col-md-4 pt-1 px-0">
                        <label htmlFor="email" className="font-weight-bold">Email address:</label>
                      </div>
                      <div className="col-md-8 px-0">
                        <Field
                          type="email"
                          className="form-control bg-white"
                          name="email"
                          id="email"
                          value={values?.email}
                          disabled
                        />
                      </div>
                    </div>
                    <div className="row m-0 pt-3">
                      <div className="col-md-4 pt-1 px-0">
                        <label htmlFor="currentPassword" className="font-weight-bold">Current password:</label>
                      </div>
                      <div className="col-md-8 px-0">
                        <Field
                          type="password"
                          className={className('form-control', { 'border-danger': errors.currentPassword })}
                          name="currentPassword"
                          id="currentPassword"
                          placeholder="Current password"
                        />
                        {touched.currentPassword && errors.currentPassword && <div className="text-danger mt-3">{errors?.currentPassword}</div>}
                      </div>
                    </div>
                    <div className="row m-0 pt-3">
                      <div className="col-md-4 pt-1 px-0">
                        <label htmlFor="newPassword" className="font-weight-bold">New password:</label>
                      </div>
                      <div className="col-md-8 px-0">
                        <Field
                          type="password"
                          className={className('form-control', { 'border-danger': errors.newPassword })}
                          name="newPassword"
                          id="newPassword"
                          placeholder="New password"
                        />
                        {touched.newPassword && errors.newPassword && <div className="text-danger mt-3">{errors?.newPassword}</div>}
                      </div>
                    </div>
                    <div className="row m-0 pt-3">
                      <div className="col-md-4 pt-1 px-0">
                        <label htmlFor="confirmNewPassword" className="font-weight-bold">Confirm password:</label>
                      </div>
                      <div className="col-md-8 px-0">
                        <Field
                          type="password"
                          className={className('form-control', { 'border-danger': errors.confirmNewPassword })}
                          name="confirmNewPassword"
                          id="confirmNewPassword"
                          placeholder="Confirm password"
                        />
                        {touched.confirmNewPassword && errors.confirmNewPassword && <div className="text-danger mt-3">{errors?.confirmNewPassword}</div>}
                      </div>
                    </div>
                    <div className="row m-0 py-3">
                      <div className="col-md-3 col-sm-4 col-6 pl-0 pr-1">
                        <Button
                          type="button"
                          className="w-100"
                          variant="secondary"
                          onClick={() => { history.push(paths.MY_PROFILE); }}
                        >
                          Cancel
                        </Button>
                      </div>
                      <div className="col-md-3 offset-md-6 col-sm-4 offset-sm-4 col-6 pl-1 pr-0">
                        <Button
                          type="submit"
                          className=" w-100"
                          variant="success"
                          disabled={!isValid || !dirty || isChangePasswordLoading
                          || errors.currentPassword || errors.newPassword
                          || errors.confirmNewPassword}
                        >
                          Save
                        </Button>
                      </div>
                    </div>
                  </div>
                </Form>
              )}
            </Formik>
            <ModalWindow
              toShow={toShowModal}
              onSubmit={handleConfirm}
              onClose={handleCloseModal}
            >Are you sure you want to change password?
            </ModalWindow>
          </WithLoading>
        </div>
      </div>
    </div>
  );
};