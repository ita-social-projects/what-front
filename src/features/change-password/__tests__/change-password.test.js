import React from 'react';
import { useSelector } from 'react-redux';
import { render, fireEvent, waitFor } from '@testing-library/react';
import { createMemoryHistory } from 'history';
import { BrowserRouter as Router } from 'react-router-dom';

import { ChangePassword } from '../change-password';
import { useActions, paths } from '@/shared/index.js';
import { newPassword } from '@models/index.js';
import { addAlert } from '@/features';

jest.mock('react-redux', () => ({
  useSelector: jest.fn().mockReturnValue({
    isLoading: false,
    error: '',
    loaded: true,
  }),
}));

jest.mock('@/shared/hooks', () => ({
  useActions: jest.fn().mockReturnValue([jest.fn(), jest.fn()]),
}));

describe('Change password', () => {
  let formValues;
  beforeEach(() => {
    formValues = {
      email: 'testmail@gmail.com',
      currentPassword: 'OldTestpassw!ord123',
      newPassword: 'Testpassw!ord123',
      confirmNewPassword: 'Testpassw!ord123',
    };
  });

  it('should render component', () => {
    const { container } = render(
      <Router>
        <ChangePassword />
      </Router>
    );
    const changePasswordContainer = container.getElementsByClassName(
      'container'
    );

    expect(changePasswordContainer).toMatchSnapshot();
  });

  it('should validate the form', async () => {
    const { getByText } = render(
      <Router>
        <ChangePassword />
      </Router>
    );

    const submitButton = getByText('Save');

    await waitFor(() => {
      fireEvent.click(submitButton);
    });

    expect(submitButton).toBeDisabled();
  });

  it('current user does not exist', async () => {
    const history = createMemoryHistory();

    useSelector.mockReturnValue({
      currentUser: null,
    });

    render(
      <Router history={history}>
        <ChangePassword />
      </Router>
    );

    history.push(paths.NOT_FOUND);

    expect(history.location.pathname).toBe(paths.NOT_FOUND);
  });

  it('should handle cancel action', async () => {
    const history = createMemoryHistory();
    const INITIAL_PATH = '/';

    const { getByText } = render(
      <Router history={history}>
        <ChangePassword />
      </Router>
    );

    const cancelButton = getByText('Cancel');

    await waitFor(() => {
      fireEvent.click(cancelButton);
    });

    expect(history.location.pathname).toBe(INITIAL_PATH);
  });

  it('should handle error receiving', () => {
    useSelector.mockReturnValue({
      isLoading: false,
      error: 'Something went wrong',
      loaded: true,
    });

    render(
      <Router>
        <ChangePassword />
      </Router>
    );

    const [_, dispatchAddAlert] = useActions([newPassword, addAlert]);

    expect(dispatchAddAlert).toHaveBeenCalled();
  });

  it('should submit the form with correct schema', async () => {
    const { getByPlaceholderText, getByText } = render(
      <Router>
        <ChangePassword />
      </Router>
    );

    const email = getByPlaceholderText('Email address');
    const currentPassword = getByPlaceholderText('Current password');
    const password = getByPlaceholderText('New password');
    const confirmNewPassword = getByPlaceholderText('Confirm password');
    const submitButton = getByText('Save');

    const [setNewPassword, _] = useActions(newPassword, addAlert);

    fireEvent.change(email, {
      target: {
        value: formValues.email,
      },
    });

    fireEvent.change(currentPassword, {
      target: {
        value: formValues.currentPassword,
      },
    });

    fireEvent.change(password, {
      target: {
        value: formValues.newPassword,
      },
    });

    fireEvent.change(confirmNewPassword, {
      target: {
        value: formValues.confirmNewPassword,
      },
    });

    await waitFor(() => {
      fireEvent.click(submitButton);
    });

    setNewPassword({
      email: email.value,
      currentPassword: currentPassword.value,
      newPassword: password.value,
      confirmNewPassword: confirmNewPassword.value,
    });

    expect(setNewPassword).toHaveBeenCalledWith(formValues);
  });
});
