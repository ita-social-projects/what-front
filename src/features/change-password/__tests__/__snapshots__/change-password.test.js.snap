// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Change password should render component 1`] = `
HTMLCollection [
  <div
    class="container card shadow col-10"
  >
    <div
      class="container pb-2"
    >
      <h3
        class="pt-3"
      >
        Change Password
      </h3>
      <hr />
      <form
        action="#"
      >
        <div
          class="container px-0"
        >
          <div
            class="row m-0"
          >
            <div
              class="col-md-4 pt-1 px-0"
            >
              <label
                class="font-weight-bold"
                for="email"
              >
                Email address:
              </label>
            </div>
            <div
              class="col-md-8 px-0"
            >
              <input
                class="form-control bg-white"
                disabled=""
                id="email"
                name="email"
                placeholder="Email address"
                type="email"
                value=""
              />
            </div>
          </div>
          <div
            class="row m-0 pt-3"
          >
            <div
              class="col-md-4 pt-1 px-0"
            >
              <label
                class="font-weight-bold"
                for="currentPassword"
              >
                Current password:
              </label>
            </div>
            <div
              class="col-md-8 px-0"
            >
              <input
                class="form-control"
                id="currentPassword"
                name="currentPassword"
                placeholder="Current password"
                type="password"
                value=""
              />
            </div>
          </div>
          <div
            class="row m-0 pt-3"
          >
            <div
              class="col-md-4 pt-1 px-0"
            >
              <label
                class="font-weight-bold"
                for="newPassword"
              >
                New password:
              </label>
            </div>
            <div
              class="col-md-8 px-0"
            >
              <input
                class="form-control"
                id="newPassword"
                name="newPassword"
                placeholder="New password"
                type="password"
                value=""
              />
            </div>
          </div>
          <div
            class="row m-0 pt-3"
          >
            <div
              class="col-md-4 pt-1 px-0"
            >
              <label
                class="font-weight-bold"
                for="confirmNewPassword"
              >
                Confirm password:
              </label>
            </div>
            <div
              class="col-md-8 px-0"
            >
              <input
                class="form-control"
                id="confirmNewPassword"
                name="confirmNewPassword"
                placeholder="Confirm password"
                type="password"
                value=""
              />
            </div>
          </div>
          <div
            class="row m-0 py-3"
          >
            <div
              class="col-md-3 col-sm-4 col-6 pl-0 pr-1"
            >
              <button
                class="btn undefined w-100"
                type="button"
              >
                Cancel
              </button>
            </div>
            <div
              class="col-md-3 offset-md-6 col-sm-4 offset-sm-4 col-6 pl-1 pr-0"
            >
              <button
                class="btn undefined  w-100"
                disabled=""
                type="submit"
              >
                Save
              </button>
            </div>
          </div>
        </div>
      </form>
    </div>
  </div>,
  <div
    class="container pb-2"
  >
    <h3
      class="pt-3"
    >
      Change Password
    </h3>
    <hr />
    <form
      action="#"
    >
      <div
        class="container px-0"
      >
        <div
          class="row m-0"
        >
          <div
            class="col-md-4 pt-1 px-0"
          >
            <label
              class="font-weight-bold"
              for="email"
            >
              Email address:
            </label>
          </div>
          <div
            class="col-md-8 px-0"
          >
            <input
              class="form-control bg-white"
              disabled=""
              id="email"
              name="email"
              placeholder="Email address"
              type="email"
              value=""
            />
          </div>
        </div>
        <div
          class="row m-0 pt-3"
        >
          <div
            class="col-md-4 pt-1 px-0"
          >
            <label
              class="font-weight-bold"
              for="currentPassword"
            >
              Current password:
            </label>
          </div>
          <div
            class="col-md-8 px-0"
          >
            <input
              class="form-control"
              id="currentPassword"
              name="currentPassword"
              placeholder="Current password"
              type="password"
              value=""
            />
          </div>
        </div>
        <div
          class="row m-0 pt-3"
        >
          <div
            class="col-md-4 pt-1 px-0"
          >
            <label
              class="font-weight-bold"
              for="newPassword"
            >
              New password:
            </label>
          </div>
          <div
            class="col-md-8 px-0"
          >
            <input
              class="form-control"
              id="newPassword"
              name="newPassword"
              placeholder="New password"
              type="password"
              value=""
            />
          </div>
        </div>
        <div
          class="row m-0 pt-3"
        >
          <div
            class="col-md-4 pt-1 px-0"
          >
            <label
              class="font-weight-bold"
              for="confirmNewPassword"
            >
              Confirm password:
            </label>
          </div>
          <div
            class="col-md-8 px-0"
          >
            <input
              class="form-control"
              id="confirmNewPassword"
              name="confirmNewPassword"
              placeholder="Confirm password"
              type="password"
              value=""
            />
          </div>
        </div>
        <div
          class="row m-0 py-3"
        >
          <div
            class="col-md-3 col-sm-4 col-6 pl-0 pr-1"
          >
            <button
              class="btn undefined w-100"
              type="button"
            >
              Cancel
            </button>
          </div>
          <div
            class="col-md-3 offset-md-6 col-sm-4 offset-sm-4 col-6 pl-1 pr-0"
          >
            <button
              class="btn undefined  w-100"
              disabled=""
              type="submit"
            >
              Save
            </button>
          </div>
        </div>
      </div>
    </form>
  </div>,
  <div
    class="container px-0"
  >
    <div
      class="row m-0"
    >
      <div
        class="col-md-4 pt-1 px-0"
      >
        <label
          class="font-weight-bold"
          for="email"
        >
          Email address:
        </label>
      </div>
      <div
        class="col-md-8 px-0"
      >
        <input
          class="form-control bg-white"
          disabled=""
          id="email"
          name="email"
          placeholder="Email address"
          type="email"
          value=""
        />
      </div>
    </div>
    <div
      class="row m-0 pt-3"
    >
      <div
        class="col-md-4 pt-1 px-0"
      >
        <label
          class="font-weight-bold"
          for="currentPassword"
        >
          Current password:
        </label>
      </div>
      <div
        class="col-md-8 px-0"
      >
        <input
          class="form-control"
          id="currentPassword"
          name="currentPassword"
          placeholder="Current password"
          type="password"
          value=""
        />
      </div>
    </div>
    <div
      class="row m-0 pt-3"
    >
      <div
        class="col-md-4 pt-1 px-0"
      >
        <label
          class="font-weight-bold"
          for="newPassword"
        >
          New password:
        </label>
      </div>
      <div
        class="col-md-8 px-0"
      >
        <input
          class="form-control"
          id="newPassword"
          name="newPassword"
          placeholder="New password"
          type="password"
          value=""
        />
      </div>
    </div>
    <div
      class="row m-0 pt-3"
    >
      <div
        class="col-md-4 pt-1 px-0"
      >
        <label
          class="font-weight-bold"
          for="confirmNewPassword"
        >
          Confirm password:
        </label>
      </div>
      <div
        class="col-md-8 px-0"
      >
        <input
          class="form-control"
          id="confirmNewPassword"
          name="confirmNewPassword"
          placeholder="Confirm password"
          type="password"
          value=""
        />
      </div>
    </div>
    <div
      class="row m-0 py-3"
    >
      <div
        class="col-md-3 col-sm-4 col-6 pl-0 pr-1"
      >
        <button
          class="btn undefined w-100"
          type="button"
        >
          Cancel
        </button>
      </div>
      <div
        class="col-md-3 offset-md-6 col-sm-4 offset-sm-4 col-6 pl-1 pr-0"
      >
        <button
          class="btn undefined  w-100"
          disabled=""
          type="submit"
        >
          Save
        </button>
      </div>
    </div>
  </div>,
]
`;

exports[`Reset password should render component 1`] = `
HTMLCollection [
  <div
    class="container card shadow col-10"
  >
    <div
      class="container pb-2"
    >
      <h3
        class="pt-3"
      >
        Change Password
      </h3>
      <hr />
      <form
        action="#"
      >
        <div
          class="container px-0"
        >
          <div
            class="row m-0"
          >
            <div
              class="col-md-4 pt-1 px-0"
            >
              <label
                class="font-weight-bold"
                for="email"
              >
                Email address:
              </label>
            </div>
            <div
              class="col-md-8 px-0"
            >
              <input
                class="form-control bg-white"
                disabled=""
                id="email"
                name="email"
                placeholder="Email address"
                type="email"
                value=""
              />
            </div>
          </div>
          <div
            class="row m-0 pt-3"
          >
            <div
              class="col-md-4 pt-1 px-0"
            >
              <label
                class="font-weight-bold"
                for="currentPassword"
              >
                Current password:
              </label>
            </div>
            <div
              class="col-md-8 px-0"
            >
              <input
                class="form-control"
                id="currentPassword"
                name="currentPassword"
                placeholder="Current password"
                type="password"
                value=""
              />
            </div>
          </div>
          <div
            class="row m-0 pt-3"
          >
            <div
              class="col-md-4 pt-1 px-0"
            >
              <label
                class="font-weight-bold"
                for="newPassword"
              >
                New password:
              </label>
            </div>
            <div
              class="col-md-8 px-0"
            >
              <input
                class="form-control"
                id="newPassword"
                name="newPassword"
                placeholder="New password"
                type="password"
                value=""
              />
            </div>
          </div>
          <div
            class="row m-0 pt-3"
          >
            <div
              class="col-md-4 pt-1 px-0"
            >
              <label
                class="font-weight-bold"
                for="confirmNewPassword"
              >
                Confirm password:
              </label>
            </div>
            <div
              class="col-md-8 px-0"
            >
              <input
                class="form-control"
                id="confirmNewPassword"
                name="confirmNewPassword"
                placeholder="Confirm password"
                type="password"
                value=""
              />
            </div>
          </div>
          <div
            class="row m-0 py-3"
          >
            <div
              class="col-md-3 col-sm-4 col-6 pl-0 pr-1"
            >
              <button
                class="btn undefined w-100"
                type="button"
              >
                Cancel
              </button>
            </div>
            <div
              class="col-md-3 offset-md-6 col-sm-4 offset-sm-4 col-6 pl-1 pr-0"
            >
              <button
                class="btn undefined  w-100"
                disabled=""
                type="submit"
              >
                Save
              </button>
            </div>
          </div>
        </div>
      </form>
    </div>
  </div>,
  <div
    class="container pb-2"
  >
    <h3
      class="pt-3"
    >
      Change Password
    </h3>
    <hr />
    <form
      action="#"
    >
      <div
        class="container px-0"
      >
        <div
          class="row m-0"
        >
          <div
            class="col-md-4 pt-1 px-0"
          >
            <label
              class="font-weight-bold"
              for="email"
            >
              Email address:
            </label>
          </div>
          <div
            class="col-md-8 px-0"
          >
            <input
              class="form-control bg-white"
              disabled=""
              id="email"
              name="email"
              placeholder="Email address"
              type="email"
              value=""
            />
          </div>
        </div>
        <div
          class="row m-0 pt-3"
        >
          <div
            class="col-md-4 pt-1 px-0"
          >
            <label
              class="font-weight-bold"
              for="currentPassword"
            >
              Current password:
            </label>
          </div>
          <div
            class="col-md-8 px-0"
          >
            <input
              class="form-control"
              id="currentPassword"
              name="currentPassword"
              placeholder="Current password"
              type="password"
              value=""
            />
          </div>
        </div>
        <div
          class="row m-0 pt-3"
        >
          <div
            class="col-md-4 pt-1 px-0"
          >
            <label
              class="font-weight-bold"
              for="newPassword"
            >
              New password:
            </label>
          </div>
          <div
            class="col-md-8 px-0"
          >
            <input
              class="form-control"
              id="newPassword"
              name="newPassword"
              placeholder="New password"
              type="password"
              value=""
            />
          </div>
        </div>
        <div
          class="row m-0 pt-3"
        >
          <div
            class="col-md-4 pt-1 px-0"
          >
            <label
              class="font-weight-bold"
              for="confirmNewPassword"
            >
              Confirm password:
            </label>
          </div>
          <div
            class="col-md-8 px-0"
          >
            <input
              class="form-control"
              id="confirmNewPassword"
              name="confirmNewPassword"
              placeholder="Confirm password"
              type="password"
              value=""
            />
          </div>
        </div>
        <div
          class="row m-0 py-3"
        >
          <div
            class="col-md-3 col-sm-4 col-6 pl-0 pr-1"
          >
            <button
              class="btn undefined w-100"
              type="button"
            >
              Cancel
            </button>
          </div>
          <div
            class="col-md-3 offset-md-6 col-sm-4 offset-sm-4 col-6 pl-1 pr-0"
          >
            <button
              class="btn undefined  w-100"
              disabled=""
              type="submit"
            >
              Save
            </button>
          </div>
        </div>
      </div>
    </form>
  </div>,
  <div
    class="container px-0"
  >
    <div
      class="row m-0"
    >
      <div
        class="col-md-4 pt-1 px-0"
      >
        <label
          class="font-weight-bold"
          for="email"
        >
          Email address:
        </label>
      </div>
      <div
        class="col-md-8 px-0"
      >
        <input
          class="form-control bg-white"
          disabled=""
          id="email"
          name="email"
          placeholder="Email address"
          type="email"
          value=""
        />
      </div>
    </div>
    <div
      class="row m-0 pt-3"
    >
      <div
        class="col-md-4 pt-1 px-0"
      >
        <label
          class="font-weight-bold"
          for="currentPassword"
        >
          Current password:
        </label>
      </div>
      <div
        class="col-md-8 px-0"
      >
        <input
          class="form-control"
          id="currentPassword"
          name="currentPassword"
          placeholder="Current password"
          type="password"
          value=""
        />
      </div>
    </div>
    <div
      class="row m-0 pt-3"
    >
      <div
        class="col-md-4 pt-1 px-0"
      >
        <label
          class="font-weight-bold"
          for="newPassword"
        >
          New password:
        </label>
      </div>
      <div
        class="col-md-8 px-0"
      >
        <input
          class="form-control"
          id="newPassword"
          name="newPassword"
          placeholder="New password"
          type="password"
          value=""
        />
      </div>
    </div>
    <div
      class="row m-0 pt-3"
    >
      <div
        class="col-md-4 pt-1 px-0"
      >
        <label
          class="font-weight-bold"
          for="confirmNewPassword"
        >
          Confirm password:
        </label>
      </div>
      <div
        class="col-md-8 px-0"
      >
        <input
          class="form-control"
          id="confirmNewPassword"
          name="confirmNewPassword"
          placeholder="Confirm password"
          type="password"
          value=""
        />
      </div>
    </div>
    <div
      class="row m-0 py-3"
    >
      <div
        class="col-md-3 col-sm-4 col-6 pl-0 pr-1"
      >
        <button
          class="btn undefined w-100"
          type="button"
        >
          Cancel
        </button>
      </div>
      <div
        class="col-md-3 offset-md-6 col-sm-4 offset-sm-4 col-6 pl-1 pr-0"
      >
        <button
          class="btn undefined  w-100"
          disabled=""
          type="submit"
        >
          Save
        </button>
      </div>
    </div>
  </div>,
]
`;
